pyramid/config/views.py

```diff
@@ -908,8 +943,41 @@ def register(permission=permission, renderer=renderer):
             # see also MultiView.__discriminator__
             view_intr['derived_callable'] = derived_view
 
-            registered = self.registry.adapters.registered
+            # make a new view separately for normal and exception paths
+            view_classifiers = []
+            if exception is None:
+                view_classifiers.append(IViewClassifier)
+            if isexc:
```

Comment:
This was confusing at first. My brain didn't immediately click that `isexc` could potentially be true when the user is registering a context that also happens to be an exception without using `exception=`. I'd recommend a comment of some sort to make it more clear what is going on so that in the future someone doesn't think `isexc` means it is an `exception` so why would `exception` be `None`.